packages/loopback/test/acceptance/routing/routing.acceptance.ts

```diff
@@ -90,6 +94,68 @@ describe('Routing', () => {
     expect(result).to.have.property('body', 'TestApp');
   });
 
+  it('creates a new child context for each request', async () => {
+    const app = givenAnApplication();
+    app.bind('flag').to('original');
+
+    const spec = givenOpenApiSpec()
+      .withOperationReturningString('put', '/flag', 'setFlag')
+      .withOperationReturningString('get', '/flag', 'getFlag')
+      .build();
+
+    @api(spec)
+    class FlagController {
+      constructor(@inject('ctx') private ctx: Context) {
```

Comment:
The reason why I don't like this approach is that user code should never access the DI container (`ctx`) directly, that breaks the whole DependencyInjection/InversionOfControl design.